lens/private/compound/join-list.rkt

```diff
@@ -1,26 +1,24 @@
-#lang racket/base
+#lang sweet-exp racket/base
 
-(require racket/list
-         racket/contract
-         "../base/main.rkt")
+require racket/list
+        racket/contract
+        "../base/main.rkt"
+        "../util/alternating-list.rkt"
 
-(module+ test
-  (require rackunit
-           "../list/list-ref-take-drop.rkt"))
-
-(provide
- (contract-out
-  [lens-join/list (->* () #:rest (listof lens?) (lens/c any/c list?))]))
+module+ test
+  require rackunit
+          "../list/list-ref-take-drop.rkt"
 
+provide
+  contract-out
+    lens-join/list (->* () #:rest (listof lens?) (lens/c any/c list?))
```

Comment:
While we're touching on this, there's a `rest->` contract combinator in `private/util` that would make this contract much simpler:

```racket
(rest-> lens? (lens/c any/c list?))
```